### DISCUSSION TITLE
Over-large window sizes do not produce a properly maximised window

### DISCUSSION THREAD

**Discussion by EnronEvolved** [ghostty-org/ghostty#3580] [Feature Requests, Ideas] [answered]
The documentation for ghostty configuration states that setting `window-width` and `window-height` to massive values will produce a maximised window, due to Ghostty's clamping behaviour. 

However, while this produces a window of the correct size, it does not produce a window in the correct maximised window state. This means that the window has rounded top corners when properly maximised windows lack them, and when you click the button to "turn off" maximisation, it instead properly maximises the window. I believe this clumsiness is antithetical to Ghostty's design goals.

This is my `ghostty --version`, running on Linux Mint 21.3, with Cinnamon 6.0.1 as desktop environment:

```
Ghostty 1.0.1-main+a8e5eef1

Version
  - version: 1.0.1-main+a8e5eef1
  - channel: tip
Build Config
  - Zig version: 0.13.0
  - build mode : builtin.OptimizeMode.ReleaseFast
  - app runtime: apprt.Runtime.gtk
  - font engine: font.main.Backend.fontconfig_freetype
  - renderer   : renderer.OpenGL
  - libxev     : main.Backend.io_uring
  - GTK version:
    build      : 4.6.9
    runtime    : 4.6.9
  - libadwaita : disabled
```

I suggest keeping `window-width` and `window-height` as the default sizes for the window when it is neither maximised nor minimised, and implementing a default state in a manner similar to Alacritty.

**Comment by 00-kat** [comment, marked as answer]
Duplicate of #3642 although I'm not sure which one was first.

**Reply by EnronEvolved** [reply]
Surely the lower number, and the earlier date, of this discussion make it obvious. It seems the later discussion is the one that took off.

**Reply by jcollie** [reply]
In any case, this appears to have been fixed in 1.1.0.

**Reply by 00-kat** [reply]
It's possible it were moved from an issue which was made earlier (I'm not sure if/where GitHub exposes that in the UI), so I did not say that it is first for certain.
